internal/analytics/analytics.go

```diff
@@ -82,11 +84,31 @@ func (d *customDimensions) toMap() map[string]string {
 	}
 }
 
+var (
+	eventWaitGroup sync.WaitGroup
+)
+
 func init() {
 	CustomDimensions = &customDimensions{}
 	setup()
 }
 
+// WaitForAllEvents waits for all events to return
+func WaitForAllEvents(t time.Duration) {
+	wg := make(chan bool)
+	go func() {
+		eventWaitGroup.Wait()
+		wg <- true
```

Comment:
nit/not-nit:
```suggestion
		defer close(wg)
		eventWaitGroup.Wait()
```
This ensures resource cleanup.

Reply:
Also, it's idiomatic to use a `chan struct{}` since `struct{}` is a zero-sized primitive.